src/screens/TopicPageScreen/index.js

```diff
@@ -74,14 +71,54 @@ const TopicPageScreen = (props) => {
   const [offset, setOffset] = React.useState(0);
   const [client] = React.useContext(Context).client;
   const refBlockComponent = React.useRef();
-  const interactionManagerRef = React.useRef(null);
-  const interactionManagerAnimatedRef = React.useRef(null);
+  const scrollY = React.useRef(new Animated.Value(0)).current;
   const {mappingColorFeed} = useCoreFeed();
   const {listRef} = useOnBottomNavigationTabPressHook(LIST_VIEW_TYPE.TIKTOK_SCROLL, onRefresh);
 
   const topicWithPrefix = route.params.id;
   const id = removePrefixTopic(topicWithPrefix);
 
+  const coverPath = topicDetail?.cover_path || null;
+  const {top} = useSafeAreaInsets();
+  const topPosition = Platform.OS === 'ios' ? top : 0;
+
+  const navigationHeight = coverPath
+    ? dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT_COVER
+    : dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT;
+  const headerShowHeight =
+    navigationHeight + dimen.size.TOPIC_FEED_HEADER_HEIGHT + topPosition + normalize(4);
+  const headerHideHeight = dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT2 + topPosition;
+
+  const headerHeight = scrollY.interpolate({
+    inputRange: [0, headerHideHeight],
+    outputRange: [headerShowHeight, headerHideHeight],
+    extrapolate: 'clamp'
+  });
+
+  const opacityHeader = scrollY.interpolate({
+    inputRange: [0, 100],
+    outputRange: [1, 0],
+    extrapolate: 'clamp'
+  });
+
+  React.useEffect(() => {
+    const listener = scrollY.addListener(({value}) => {
+      if (value >= 0 && value <= headerHideHeight) {
+        setIsHeaderHide(false);
+      } else if (value >= headerHideHeight) {
+        setIsHeaderHide(true);
+      }
+    });
+
+    return () => {
+      scrollY.removeListener(listener);
+    };
+  }, [scrollY]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `scrollY` variable and related logic have been added. This seems to be a replacement for the removed `animatedHeight` and `opacityHeaderAnimation`. Ensure that this new implementation works as expected.
```diff
+   const scrollY = React.useRef(new Animated.Value(0)).current;
+   const coverPath = topicDetail?.cover_path || null;
+   const {top} = useSafeAreaInsets();
+   const topPosition = Platform.OS === 'ios' ? top : 0;
+ 
+   const navigationHeight = coverPath
+     ? dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT_COVER
+     : dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT;
+   const headerShowHeight =
+     navigationHeight + dimen.size.TOPIC_FEED_HEADER_HEIGHT + topPosition + normalize(4);
+   const headerHideHeight = dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT2 + topPosition;
+ 
+   const headerHeight = scrollY.interpolate({
+     inputRange: [0, headerHideHeight],
+     outputRange: [headerShowHeight, headerHideHeight],
+     extrapolate: 'clamp'
+   });
+ 
+   const opacityHeader = scrollY.interpolate({
+     inputRange: [0, 100],
+     outputRange: [1, 0],
+     extrapolate: 'clamp'
+   });
+ 
+   React.useEffect(() => {
+     const listener = scrollY.addListener(({value}) => {
+       if (value >= 0 && value <= headerHideHeight) {
+         setIsHeaderHide(false);
+       } else if (value >= headerHideHeight) {
+         setIsHeaderHide(true);
+       }
+     });
+ 
+     return () => {
+       scrollY.removeListener(listener);
+     };
+   }, [scrollY]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->